src/components/map/rent-popup.tsx

```diff
@@ -87,6 +87,23 @@ const RentPopup: React.FC<RentPopupProps> = ({
       });
   }, [booking, fetchBooking, fetchStationDetail, selectedStation, BUCHUNGEN]);
 
+  const handleRefreshBooking = useCallback(() => {
+    if (!booking) {
+      throw new Error('Trying to refresh a booking, but no bike booked.');
+    }
+    if (!selectedStation) {
+      throw new Error('Trying to refresh a booking, but no station selected.');
+    }
+
+    return cancelBooking()
+      .then(() => bookBike(selectedStation.stationId))
+      .then(() => Promise.all([fetchBooking(), fetchStationDetail()]))
+      .catch(err => {
+        console.error('Error while refreshing a booking:', err);
+        toast(BUCHUNGEN.ALERT.LOAD_CURR_BOOKING_ERR, { type: 'error' });
+      });
+  }, [booking, fetchBooking, selectedStation]);
```

Comment:
Please also add `fetchStationDetail` to the hook dependencies.

Reply:
See efc85890c81a2ed0013fe20804acade150ca5ea8.